sdk/typescript/src/experimental/types.ts

```diff
@@ -0,0 +1,253 @@
+ *   ObjectID: '4047d2e25211d87922b6650233bd0503a6734279',
+ *   SequenceNumber: 1,
+ *   ObjectDigest: Buffer.from('bCiANCht4O9MEUhuYjdRCqRPZjr2rJ8MfqNiwyhmRgA=', 'base64')
+ * } };
+ * ```
+ *
+ * For `Pure` arguments BCS is required. You must encode the values with BCS according
+ * to the type required by the called function. Pure accepts only serialized values
+ */
+export type CallArg =
+  | { SharedObject: string }
+  | { Pure: Iterable<number> }
+  | { ImmOrOwnedObject: SuiObjectRef }
+
+initializers.push((b: typeof BCS) =>
+  b.registerEnumType("CallArg", {
+    Pure: "vector<u8>",
+    ImmOrOwnedObject: "SuiObjectRef",
+    SharedObject: "ObjectID",
+  })
+);
+
+export type StructTag = {
+  address: string,
+  module: string,
+  name: string,
+  typeParams: TypeTag[]
+};
+
+export type TypeTag =
+  | { bool: null }
+  | { u8: null }
+  | { u64: null }
+  | { u128: null }
+  | { address: null }
+  | { signer: null }
+  | { vector: TypeTag }
+  | { struct: StructTag }
+
+initializers.push((b: typeof BCS) => b
+  .registerEnumType('TypeTag', {
+    bool: null,
+    u8: null,
+    u64: null,
+    u128: null,
+    address: null,
+    signer: null,
+    vector: 'TypeTag',
+    struct: 'StructTag'
+  })
+  .registerVectorType('vector<TypeTag>', 'TypeTag')
+  .registerStructType('StructTag', {
+    address: 'string',
+    module: 'string',
+    name: 'string',
+    typeParams: 'vector<TypeTag>'
+  })
+);
+
+/**
+ * Transaction type used for calling Move modules' functions.
+ * Should be crafted carefully, because the order of type parameters and
+ * arguments matters.
+ */
+export type MoveCallTx = {
+  Call: {
+    package: SuiObjectRef;
+    module: string;
+    function: string;
+    typeArguments: TypeTag[];
+    arguments: CallArg[];
+  };
+}
+
+initializers.push((b: typeof BCS) =>
+  b
+    .registerVectorType("vector<CallArg>", "CallArg")
+    .registerStructType("MoveCallTx", {
+      package: "SuiObjectRef",
+      module: "string",
+      function: "string",
+      typeArguments: "vector<TypeTag>",
+      arguments: "vector<CallArg>",
+    })
+);
+
+// ========== TransactionData ===========
+
+export type Transaction = MoveCallTx | PublishTx | TransferCoinTx;
+
+initializers.push((b: typeof BCS) =>
+  b.registerEnumType("Transaction", {
+    TransferCoin: "TransferCoinTx",
+    Publish: "PublishTx",
+    Call: "MoveCallTx",
+  })
+);
+
+export type TransactionKind =
+  | { Single: Transaction }
+  | { Batch: Transaction[] };
+
+initializers.push((b: typeof BCS) =>
+  b
+    .registerVectorType("vector<Transaction>", "Transaction")
+    .registerEnumType("TransactionKind", {
+      Single: "Transaction",
+      Batch: "vector<Transaction>",
+    })
+);
+
+/**
+ * The TransactionData to be signed and sent to the Gateway service.
+ *
+ * Field `sender` is made optional as it can be added during the signing
+ * process and there's no need to define it sooner.
+ */
+export type TransactionData = {
```

Comment:
what's the difference between this and the regular `TransactionData` that's on main already ?
I see some minor formatting changes but the purpose and data seem to largely be the same.